explore cell tests

cryogenian opened the pull request:
fixes #267 
slamengine.jar updated to 2.1.8
added option to not launch test slamengine jar.

garyb commented on test/config.json:
```diff
@@ -9,7 +9,7 @@
         "waitTime": 6000
     },
-    "slamdataUrl": "http://localhost:63175",
+    "slamdataUrl": "http://local.slamdata.com",
```
Can you revert that? :wink:

Perhaps we should set up a local-machine override file that we can include if present, and add to the `.gitignore` so this kind of thing doesn't happen.

garyb commented:
What's the advantage of not using the included slamengine by the way? Seems like it might be asking for trouble if we're not all running the tests with the same configuration.

cryogenian commented:
It doesn't work with static files served by slamengine for me. So, I point to my local slamdata and turn slamengine from test folder off.

garyb commented:
That sounds like something that should probably be fixed then? :wink: Is there a slamengine issue for that?

cryogenian commented:
No, this bug isn't fixed :disappointed: 
I'll make issue in slamdata because I've no evidence slamengine works incorrect (it maybe selenium, firefox, MacOS) and it works for everyone but me. 

* reverted `slamdataUrl`
* removed `ignoreEngine` option

garyb commented:
Oh you didn't have to remove the option, I just wondered about the reasoning behind it! If you can't test without it being a pain currently it's definitely worth keeping.

cryogenian commented:
Nevermind, I can test without this option :smile:

jdegoes commented:
@garyb Happy with this?

garyb commented:
Looks good to me, my only other comment would be is there any way we can avoid some or all of the `waitTime`s in favour of checking some condition instead? Anything to help reduce nondeterminism in these tests!

@jonsterling have any comments too?

jdegoes commented:
> Looks good to me, my only other comment would be is there any way we can avoid some or all of the waitTimes in favour of checking some condition instead?

Agreed, these wait times *need to go*. They mean the tests will *definitely fail some percentage of the time*. The conditions should be checked for continuously until success (or until a large timeout, e.g. 2 minutes, which means they will never false positive and you only pay the timeout cost when the code is actually broken).

jonsterling commented:
Looks good!

I would just echo @garyb in that I think it would be great if we could avoid all these `waitTime` calls. When running on our own machines, it may work most of the time by coincidence, but it you tried running it on a much slower machine (such as through Sauce Labs, for instance), we may begin to have more false negatives.

Fwiw, I was planning to try and replace most delays with `waitCheck` calls on real conditions in the `File` component tests soon (perhaps today).

cryogenian commented:
Ok, is it acceptable if I replace those `waitTime` during work on #266?

jdegoes commented:
> Ok, is it acceptable if I replace those waitTime during work on #266?

Up to @jonsterling and @garyb. :smile:

jonsterling commented:
@cryogenian There might be an argument for merging now (after the merge conflict is fixed), rather than waiting until the `waitTime`s are removed—since we're working on the tests a fair bit now and it would be a shame if further merge conflicts made this one more difficult.

I don't feel too strongly about whether the `waitTime`s are removed now, or in the very near future; it's also something I'm happy to help out on. I'll leave the final recommendation up to @garyb.

garyb commented:
Yep, merging now is fine by me if we're going to re-look at the `waitTimes` soonish :)

cryogenian commented:
rebased
